### PR TITLE
Cleanup ViewMode string handling

### PR DESCRIPTION
**Pull request by jeremypw**
Use ViewMode enum functions instead of hard-coded strings in Window and ViewSwitcher.

### CODE REVIEW

File: libcore/Enums.vala [hunk 1]
```diff
@@ -60,16 +60,46 @@ namespace Marlin {
...
         /* First three modes must match the corresponding mode switch indices */
         ICON = 0,
         LIST = 1,
         MILLER_COLUMNS = 2,
         CURRENT,
         PREFERRED,
-        INVALID
+        INVALID;
+
+        public string to_string () {
```
Comment (tintou) [inline libcore/Enums.vala]:
```suggestion
        public unowned string to_string () {
```

**Comment by jeremypw** [comment]
It occurs to me that, if we bump the minimum gtk3 version to 3.22.25  (its currently 3.22.30 in Juno), then we do not need to use a string type action target anymore, we can use the enum value as a target. I'll try it and see.

File: libcore/Enums.vala (hunk 1 above)
Comment (jeremypw) [reply libcore/Enums.vala]: OK, done.

**Comment by jeremypw** [comment]
See #1094 for a better solution.

**Pull request closed without merging**